src/native/eventpipe/ep-config.c

```diff
@@ -391,8 +391,8 @@ ep_config_build_event_metadata_event (
 	const ep_char16_t *provider_name_utf16 = ep_provider_get_provider_name_utf16 (provider);
 	const uint8_t *payload_data = ep_event_get_metadata (source_event);
 	uint32_t payload_data_len = ep_event_get_metadata_len (source_event);
-	uint32_t provider_name_len = (uint32_t)((ep_rt_utf16_string_len (provider_name_utf16) + 1) * sizeof (ep_char16_t));
-	uint32_t instance_payload_size = sizeof (metadata_id) + provider_name_len + payload_data_len;
+	uint32_t provider_name_len = (uint32_t)ep_rt_utf16_string_len (provider_name_utf16);
+	uint32_t instance_payload_size = 0;
 
 	// Allocate the payload.
 	instance_payload = ep_rt_byte_array_alloc (instance_payload_size);
```

Comment:
This is not yet calculated, it was done before in the old code, still need to be done before, since you need the buffer to serialize stuff into.

Reply:
Thought I could refactor this quickly so that we don't need to book-keep the length. Of course, this doesn't work out since we need the size a priori. Fixed in the latest push.